llpc/context/llpcCompiler.cpp

```diff
@@ -1235,13 +1234,15 @@ Result Compiler::buildPipelineInternal(Context *context, ArrayRef<const Pipeline
           timerProfiler.getTimer(TimerCodeGen),
       };
 
-      pipeline->generate(std::move(pipelineModule), elfStream, checkShaderCacheFunc, timers, cl::NewPassManager == 2);
-#if LLPC_ENABLE_EXCEPTION
-      result = Result::Success;
-#endif
+      if (pipeline->generate(std::move(pipelineModule), elfStream, checkShaderCacheFunc, timers,
+                             cl::NewPassManager == 2))
+        result = Result::Success;
+      else
+        result = Result::ErrorInvalidShader;
```

Comment:
I don't think this is right. generate() returning false is for a _recoverable_ error where the front-end might want to retry but this time doing a whole pipeline compile. For an unrecoverable error, lgc or llvm would call llvm::report_fatal_error() and the resulting exception is caught here.

Reply:
But LLPC_ENABLE_EXCEPTION is disabled now, How to return a error message?

Reply:
Where do you want to return the error message to? Currently, without LLPC_ENABLE_EXCEPTION, fatalErrorHandler outputs the error message to LLPC_ERRS, then returns to LLVM which terminates the process. If you want to be able to return from LLPC in an error condition, then LLPC_ENABLE_EXCEPTION needs to be on. If we're compiling without exceptions, maybe it should use setjmp/longjmp instead.